### ISSUE TITLE
FunctionComment.SpacingAfterParamType not working with intersection types

### ISSUE THREAD

**Issue by icanhazstring** [squizlabs/PHP_CodeSniffer#2887]
Hi there,

the given sniff `Squiz.Commenting.FunctionComment.SpacingAfterParamType` does not work well with intersection type annotations.

Used Version: 3.5.4

It might be related to an older, already fixed issue:
https://github.com/squizlabs/PHP_CodeSniffer/issues/2530

Given the following code
```
/**
 * @param ContainerInterface&ServiceManager $container
 * @param array<string, string> $options
 */
public function __invoke(ContainerInterface $container, array $options = []) : Fubar
```

Expected:
no problem

Actual:
```
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 XX | ERROR | [x] Expected 4 spaces after parameter type; 0 found
      (Squiz.Commenting.FunctionComment.SpacingAfterParamType)
```

Using `phpcbf` to fix the issue resulted in this line:
```
* @param ContainerInterface    &ServiceManager $container
* @param array<string, string> $options
```

**Comment by VincentLanglet** [comment]
@icanhazstring It's related to https://github.com/squizlabs/PHP_CodeSniffer/issues/2806

It was fixed for the PEAR standard but not the Squizz one.

Do you want to create the PR ?

I recommend 
- To create a constant or protected method in the src/Standards/PEAR/Sniffs/Commenting/FunctionCommentSniff.php
- To use this constant/method in the src/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php (which is extending the PEAR one)

This way this won't happen again if we update the regex of the PEAR Sniff.

**Comment by rodrigoprimo** [comment]
I just tested, and I can confirm that https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/944 fixed this problem.

As stated in the issue description, before https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/944, `phpcbf` would fix the code sample provided here to:

```php
/**
 * @param ContainerInterface    &ServiceManager $container
 * @param array<string, string> $options
 */
public function __invoke(ContainerInterface $container, array $options = []) : Fubar
```

Now it fixes it to:

```php
/**
 * @param ContainerInterface&ServiceManager $container
 * @param array<string, string>             $options
 */
public function __invoke(ContainerInterface $container, array $options = []) : Fubar
```

> Expected:
> no problem

`phpcs` still flags the problem, and I believe this is the correct behavior, contrary to what was said in the issue description:

```
$ bin/phpcs --standard=Squiz --sniffs=Squiz.Commenting.FunctionComment test.php -s                 

FILE: test.php
-------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-------------------------------------------------------------------------------------------------------------------------------------------------------------
 4 | ERROR | [x] Expected 4 spaces after parameter type; 0 found (Squiz.Commenting.FunctionComment.SpacingAfterParamType)
-------------------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-------------------------------------------------------------------------------------------------------------------------------------------------------------
```

I suggest closing this issue.
